src/automatic-annotations/shared/column-renderers/TaxonomicScope.tsx

```diff
@@ -0,0 +1,58 @@
+import { ExpandableList } from 'franklin-sites';
+
+import TaxonomyView from '../../../shared/components/entry/TaxonomyView';
+
+import { ARBAAPIModel } from '../../arba/adapters/arbaConverter';
+import { UniRuleAPIModel } from '../../unirule/adapters/uniRuleConverter';
+import { TaxonomyDatum } from '../../../supporting-data/taxonomy/adapters/taxonomyConverter';
+
+import helper from '../../../shared/styles/helper.module.scss';
+
+type CustomTaxonScope = {
+  negative: boolean;
+  taxon: TaxonomyDatum;
+};
+
+const TaxonomicScope = ({
+  mainRule,
+}: Partial<UniRuleAPIModel | ARBAAPIModel>) => {
+  // Using a map in order to remove duplicates
+  const taxonScopeMap = new Map<number, CustomTaxonScope>();
+  for (const conditionSet of mainRule?.conditionSets || []) {
+    for (const condition of conditionSet.conditions || []) {
+      if (condition.type !== 'taxon') {
+        continue; // eslint-disable-line no-continue
+      }
+      for (const conditionValue of condition.conditionValues || []) {
+        // This shouldn't happen
```

Comment:
Would this imply something is wrong with the data? Would a `console.warn` or `console.assert` be useful in that case?

Reply:
Not sure if that ever happens to be honest, but it's to do type guarding for TypeScript to be happy and avoid possible bugs if it does happen. I added the console.warn as suggested